### PR TITLE
Update `<SelectArrayInput>` to use default record representation when used inside `<ReferenceArrayInput>`

### PR DESCRIPTION
**Pull request by djhi**
This input didn't leverage `recordRepresentation` for the choices texts.

### CODE REVIEW

**Review by fzaninotto — APPROVED** [review]

File: packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx [hunk 1]
```diff
@@ -337,3 +327,194 @@ export const TranslateChoice = () => {
...
+            first_name: params.data.first_name,
+            last_name: params.data.last_name,
+            language: params.data.language,
+        };
+        authors.push(newAuthor);
+        return Promise.resolve({ data: newAuthor });
+    },
+} as any;
+
+export const InsideReferenceInput = () => (
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx:391]:
```suggestion
export const InsideReferenceArrayInput = () => (
```

File: packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx [hunk 2]
```diff
@@ -337,3 +327,194 @@ export const TranslateChoice = () => {
...
+                            <FormInspector name="authors" />
+                        </SimpleForm>
+                    </Edit>
+                )}
+            />
+        </Admin>
+    </TestMemoryRouter>
+);
+
+export const InsideReferenceInputDefaultValue = ({
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx:427]:
```suggestion
export const InsideReferenceArrayInputDefaultValue = ({
```

File: packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx [hunk 3]
```diff
@@ -337,3 +327,194 @@ export const TranslateChoice = () => {
...
+                            <FormInspector name="authors" />
+                        </SimpleForm>
+                    </Edit>
+                )}
+            />
+        </Admin>
+    </TestMemoryRouter>
+);
+
+export const InsideReferenceInputWithError = () => (
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx:478]:
```suggestion
export const InsideReferenceArrayInputWithError = () => (
```

File: packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx [hunk 4]
```diff
@@ -11,6 +11,8 @@ import {
     DifferentIdTypes,
     TranslateChoice,
     InsideArrayInput,
+    InsideReferenceInput,
+    InsideReferenceInputDefaultValue,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx:15]:
```suggestion
    InsideReferenceArrayInput,
    InsideReferenceArrayInputDefaultValue,
```

File: packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx [hunk 5]
```diff
@@ -667,4 +669,25 @@ describe('<SelectArrayInput />', () => {
         fireEvent.click(screen.getByLabelText('Add'));
         expect(await screen.findAllByText('Foo')).toHaveLength(2);
     });
+
+    describe('inside ReferenceInput', () => {
+        it('should use the recordRepresentation as optionText', async () => {
+            render(<InsideReferenceInput />);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx:675]:
```suggestion
            render(<InsideReferenceArrayInput />);
```

File: packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx [hunk 6]
```diff
@@ -667,4 +669,25 @@ describe('<SelectArrayInput />', () => {
...
+            await screen.findByText('Leo Tolstoy');
+        });
+        it('should not change an undefined value to empty string', async () => {
+            const onSuccess = jest.fn();
+            render(<InsideReferenceInputDefaultValue onSuccess={onSuccess} />);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx:680]:
```suggestion
            render(<InsideReferenceArrayInputDefaultValue onSuccess={onSuccess} />);
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]
Could you also update the `<SelectArrayInput>` doc? We currently have the following in `<SelectInput>`:

image

**Comment by djhi** [comment]
> Could you also update the `<SelectArrayInput>` doc? We currently have the following in `<SelectInput>`:
> 
> image

Actually, the documentation already mentioned the `recordRepresentation` even though it wasn't implemented ^^

File: packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx [hunk 7]
```diff
@@ -667,4 +669,27 @@ describe('<SelectArrayInput />', () => {
         fireEvent.click(screen.getByLabelText('Add'));
         expect(await screen.findAllByText('Foo')).toHaveLength(2);
     });
+
+    describe('inside ReferenceInput', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.spec.tsx:673]:
```suggestion
    describe('inside ReferenceArrayInput', () => {
```

**Review by fzaninotto — APPROVED** [review]

**Review by adguernier — APPROVED** [review]

**Pull request merged**